main.js

```diff
@@ -12,9 +12,42 @@ window.addEventListener('load', trigger.bind(null, 'load'));
 window.addEventListener('load', trigger.bind(null, 'DOMContentLoaded'));
 document.addEventListener('DOMContentLoaded', trigger.bind(null, 'DOMContentLoaded'));
 
+document.onreadystatechange = function () {
+	if (document.readyState === 'complete') {
+		trigger('DOMContentLoaded');
+		trigger('load');
+	}
+};
+
 if (document.readyState === 'complete') {
-	trigger('load');
-	trigger('DOMContentLoaded');
-} else if (document.readyState === 'interactive') {
```

Comment:
OK, thanks for this explanation, that makes sense. How about this?

```
--  } else if (document.readyState === 'interactive') {
++ } else if (document.readyState === 'interactive' && document.body && document.body.hasChildNodes()) {
```

If the problem is simply that document.body is undefined or empty when interactive is triggered, then this will skip firing our events at that stage.  The other code you've added is more of an optimisation for older browsers where I'm less concerned about delaying domcontentloaded until the same moment as load, because it will cease to be a problem when those browsers no longer pass CTM (in fact, isn't that already the case?  Or is this problem evident still in IE >= 9?)

So I'd be happy with this as a minimal solution that solves the problem in all browsers, with less cruft and is optimised for modern browsers.